Build failure on sparc64 with obj.as<bottom>

I recently updated Debian's packaging to the 1.4.2 release and ran into this build failure on sparc64.  All other architectures [built fine].

```
object_with_zone.cpp: In member function 'virtual void object_with_zone_user_defined_non_virtual_Test::TestBody()':
object_with_zone.cpp:745:32: error: '*((void*)& v +16)' may be used uninitialized in this function [-Werror=maybe-uninitialized]
     bottom br = obj.as<bottom>();
                                ^
object_with_zone.cpp:745:32: error: '*((void*)& v +12)' may be used uninitialized in this function [-Werror=maybe-uninitialized]
object_with_zone.cpp:745:32: error: '*((void*)& v +8)' may be used uninitialized in this function [-Werror=maybe-uninitialized]
object_with_zone.cpp:745:32: error: '*((void*)& v +4)' may be used uninitialized in this function [-Werror=maybe-uninitialized]
object_with_zone.cpp:745:32: error: 'v' may be used uninitialized in this function [-Werror=maybe-uninitialized]
```

using this toolchain

```
Kernel: Linux 4.6.0-1-sparc64-smp sparc64 (sparc64)
Toolchain package versions: binutils_2.26-9+sparc64.2 dpkg-dev_1.18.7 g++-6_6.1.1-6 gcc-6_6.1.1-6
```

[built fine]: https://buildd.debian.org/status/logs.php?pkg=msgpack-c&ver=1.4.2-1&suite=experimental

--- a/test/object_with_zone.cpp
+++ b/test/object_with_zone.cpp
@@ -742,7 +742,14 @@ TEST(object_with_zone, user_defined_non_virtual)
 
     msgpack::zone z;
     msgpack::object obj(b, z);
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif // (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) && !defined(__clang__)
     bottom br = obj.as<bottom>();
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif // (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) && !defined(__clang__)
     EXPECT_EQ(b.b, br.b);
     EXPECT_EQ(b.m1, br.m1);
     EXPECT_EQ(b.m2, br.m2);
@@ -780,7 +787,14 @@ TEST(object_with_zone, user_defined_virtual)
 
     msgpack::zone z;
     msgpack::object obj(b, z);
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif // (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) && !defined(__clang__)
     v_bottom br = obj.as<v_bottom>();
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif // (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) && !defined(__clang__)
     EXPECT_EQ(b.b, br.b);
     EXPECT_EQ(b.m1, br.m1);
     EXPECT_EQ(b.m2, br.m2);

